### ISSUE TITLE
busybox is not reproducible

### ISSUE THREAD

**Issue by osresearch** [osresearch/heads#91]
This might be caused by the musl-libc having a build path in it.

**Comment by osresearch** [comment]
The only files that still include build paths are:

```
diamond:/build/heads: find install -type f | xargs strings -t x -f | grep build.heads
install/lib/libc.so:   8b3a3 /build/heads/build/musl-cross/gcc-5.3.0/build2/x86_64-linux-musl/libgcc
install/lib/libmbedcrypto.so.0:   4db70 /build/heads/build/musl-cross/gcc-5.3.0/build2/x86_64-linux-musl/libgcc
install/lib/musl-gcc.specs:      33 -nostdinc -isystem /build/heads/install/include -isystem include%s %(old_cpp_options)
install/lib/musl-gcc.specs:      90 %(cc1_cpu) -nostdinc -isystem /build/heads/install/include -isystem include%s
install/lib/musl-gcc.specs:      ed -L/build/heads/install/lib -L .%s
install/lib/musl-gcc.specs:     14c %{!shared: /build/heads/install/lib/%{pie:S}crt1.o} /build/heads/install/lib/crti.o %{shared|pie:crtbeginS.o%s;:crtbegin.o%s}
install/lib/musl-gcc.specs:     1d5 %{shared|pie:crtendS.o%s;:crtend.o%s} /build/heads/install/lib/crtn.o
install/bin/musl-gcc:       a exec "${REALGCC:-/build/heads/crossgcc/x86_64-linux-musl/bin/x86_64-musl-linux-gcc}" "$@" -specs "/build/heads/install/lib/musl-gcc.specs"
```

**Comment by osresearch** [comment]
Those build paths are in the debug portion of the libraries, so they may not matter. After stripping the libraries to install them in `initrd`, only two files remain with diffs:

```
--- /tmp/b	2017-01-28 13:47:49.243138553 -0500
+++ /tmp/t	2017-01-28 13:47:43.283126899 -0500
@@ -19,9 +19,9 @@
 b30541afae98d8575e9eeb87aa77e82d  initrd/bin/unseal-key
 e772d074e4bec52b53f49ae1bbd00113  initrd/bin/base32
 676c832ebf2ffdb1b213cc92bad93034  initrd/bin/totp
-8b08c60d2128364d2dfd4755e0a2ee83  initrd/bin/cbmem
+d75cbfeed07fbaa28ef917a322a0218c  initrd/bin/cbmem
 ac472cf2c0de5b254a7cb2c916b4811f  initrd/bin/cryptsetup
-7b423ec6d99c14cb747b00335504ba71  initrd/bin/busybox
+6aee6634e374c99b04c9c459caec84da  initrd/bin/busybox
 83ca0a0de83f0cfdbee3accdb21bb476  initrd/bin/veritysetup
 90966d9943b78f54d486c228cfdbe56a  initrd/bin/wrap-cpio
 0549e8ee0c6122af50be0eca9b58831b  initrd/bin/kexec
```

**Comment by osresearch** [comment]
The `cbmem` diff was spurious due to an old build.

**Comment by osresearch** [comment]
busybox has build times in its executable:
```
--- /tmp/b      2017-01-28 13:58:43.650975378 -0500
+++ /tmp/t      2017-01-28 13:58:46.754968812 -0500
@@ -19175,7 +19175,7 @@
 0004ae60: 61 74 65 20 27 25 73 27 00 6f 75 74 20 6f 66 20  ate '%s'.out of 
 0004ae70: 6d 65 6d 6f 72 79 00 42 75 73 79 42 6f 78 20 76  memory.BusyBox v
 0004ae80: 31 2e 32 35 2e 30 20 28 32 30 31 37 2d 30 31 2d  1.25.0 (2017-01-
-0004ae90: 32 38 20 31 33 3a 32 31 3a 30 31 20 45 53 54 29  28 13:21:01 EST)
+0004ae90: 32 38 20 31 33 3a 34 32 3a 35 30 20 45 53 54 29  28 13:42:50 EST)
 0004aea0: 00 2f 64 65 76 2f 00 4c 49 4e 45 53 00 43 4f 4c  ./dev/.LINES.COL
 0004aeb0: 55 4d 4e 53 00 77 61 69 74 70 69 64 00 63 61 6e  UMNS.waitpid.can
 0004aec0: 27 74 20 6f 70 65 6e 20 27 25 73 27 00 63 61 6e  't open '%s'.can
```

**Comment by osresearch** [comment]
Reported to busybox: https://bugs.busybox.net/show_bug.cgi?id=9626

Patched around in 8d912ced8fef1b89687134726a722530fd0da812

**Comment by osresearch** [comment]
Busybox dev fixed it in their git tree; will be in next release. https://git.busybox.net/busybox/commit/?id=eba7fe6bb9fdc89cf9d4d33043cd3856253d303d
